tests/next_tests/unit_tests/test_common.py

```diff
@@ -64,3 +82,144 @@ def test_unit_range_index_error(rng):
+    assert UnitRange(5, 15) in domain
+    assert UnitRange(20, 30) in domain
+    assert UnitRange(25, 28) in domain
+
+
+def test_domain_contains_element(domain):
+    assert 0 in domain
+    assert 5 in domain
+    assert 20 in domain
+    assert 9 in domain
+    assert 14 in domain
+
+
+def test_domain_not_contains_range(domain):
+    assert UnitRange(25, 35) not in domain
+    assert UnitRange(10, 16) not in domain
+
+
+def test_domain_not_contains_element(domain):
+    assert 100 not in domain
+    assert -5 not in domain
+    assert 16 not in domain
+
+
+def test_positive_infinity_range():
+    pos_inf_range = UnitRange(Infinity.positive(), Infinity.positive())
+    assert len(pos_inf_range) == 0
+
+
+def test_mixed_infinity_range():
+    mixed_inf_range = UnitRange(Infinity.negative(), Infinity.positive())
+    with pytest.raises(ValueError):
+        len(mixed_inf_range)
+
+
+def test_domain_iter_method(domain):
+    iterated_values = [val for val in domain]
+    assert iterated_values == list(zip(domain.dims, domain.ranges))
+
+
+def test_domain_and_operation_different_dimensions(domain):
+    dimensions = (IDim, JDim)
+    ranges = (UnitRange(2, 12), UnitRange(7, 17))
+    domain2 = Domain(dimensions, ranges)
+    result_domain = domain & domain2
+
+    assert len(result_domain) == 3
+    assert all(isinstance(r, UnitRange) for r in result_domain.ranges)
+    assert result_domain.dims == [IDim, JDim, KDim]
+
+    assert result_domain.ranges[0] == UnitRange(2, 10)  # Intersection of domain.range1 and domain2.range1
+    assert result_domain.ranges[1] == UnitRange(7, 15)  # Intersection of domain.range2 and domain2.range2
+    assert result_domain.ranges[2] == UnitRange(20, 30)  # Broadcasting on missing dimension
+
+
+def test_domain_and_operation_different_dimensions_reversed(domain):
+    dimensions = (JDim, IDim)
+    ranges = (UnitRange(2, 12), UnitRange(7, 17))
+    domain2 = Domain(dimensions, ranges)
+
+    with pytest.raises(ValueError,
+                       match="Dimensions can not be promoted. The following dimensions appear in contradicting order: IDim, JDim."):
+        domain & domain2
+
+
+@pytest.mark.parametrize("index, expected", [
+    (0, (IDim, UnitRange(0, 10))),
+    (1, (JDim, UnitRange(5, 15))),
+    (2, (KDim, UnitRange(20, 30))),
+])
+def test_getitem_integer_index(domain, index, expected):
+    result = domain[index]
+    assert result == expected
+
+
+@pytest.mark.parametrize("slice_obj, expected", [
+    (slice(0, 2), [(IDim, UnitRange(0, 10)), (JDim, UnitRange(5, 15))]),
+    (slice(1, None), [(JDim, UnitRange(5, 15)), (KDim, UnitRange(20, 30))]),
+])
+def test_getitem_slice(domain, slice_obj, expected):
+    result = domain[slice_obj]
+    assert len(result) == len(expected)
+    assert all(res == exp for res, exp in zip(result, expected))
+
+
+import pytest
+
+
+# Assume you have already defined the Domain class and IDim, JDim, KDim classes.
+
+@pytest.fixture
+def domain_repeat_dims():
```

Comment:
I'd keep this test and check for the exception and document here that we might change it in the future